release/e2e.py

```diff
@@ -1357,17 +1356,71 @@ def _run(logger):
                     session_options["build_id"] = build_id
                     session_options["uses_app_config"] = True
 
-                if not test_config["run"].get("use_connect"):
-                    session_id = create_and_wait_for_session(
+                # Start session
+                session_id = create_and_wait_for_session(
+                    sdk=sdk,
+                    stop_event=stop_event,
+                    session_name=session_name,
+                    session_options=session_options,
+                )
+
+            prepare_command = test_config["run"].get("prepare")
+
+            # Write test state json
+            test_state_file = os.path.join(local_dir, "test_state.json")
+            with open(test_state_file, "wt") as f:
+                json.dump({
+                    "start_time": time.time(),
+                    "test_name": test_name
+                }, f)
+
+            if prepare_command or not test_uses_ray_connect:
```

Comment:
Why do we need to wrap this entire block in this criteria? In other words, why do we explicitly not want do the sync logic when  using Ray Connect without a Prepare Command?

Reply:
Previously we didn't sync anything to the Ray session when using connect. The reason why we sync with a prepare command is that we run the prepare command via the SDK API (and not connect). So far the anyscale connect tests are independent of files shipped to the cluster, so if this is feasible we should continue to test this as the minimal case.